Todo_app/services/todoAppBackend/Routers/TaskRouter.js

```diff
@@ -0,0 +1,64 @@
+const express = require("express");
+const {
+  getAllTasks,
+  updateTaskName,
+  updateTaskComplete,
+  addNewTaskToDB,
+  deleteByID,
+  handleDeleteDoneTasks,
+} = require("../mongoDB/DBFunctions/dbFunctions");
+const taskRouter = express.Router();
+
+taskRouter.get("/", async (req, res) => {
+  const doShowAll = req.query.doShowAll === "true";
+  const doSortAZ = req.query.doSortAZ === "true";
+  let tasks = await getAllTasks();
+  if (!doShowAll) {
+    tasks = tasks.filter((task) => task.complete !== true);
+  }
+  if (doSortAZ) {
+    tasks.sort(function (a, b) {
+      return a.name.localeCompare(b.name);
+    });
+  }
+  res.json(tasks);
+});
+
+taskRouter.put("/updateName/:id", async (req, res) => {
+  const body = req.body;
+  const id = req.params.id;
+  const newTaskName = body.name;
+  let data = await updateTaskName(id, newTaskName);
+  res.send(data);
+});
+
+taskRouter.put("/updateComplete/:id", async (req, res) => {
+  const body = req.body;
```

Comment:
const { taskCompleteChanged } = req.body  
another way to write it. this is good soulotion for many params/body variables